### PR TITLE
docs: add juptyer API tutorial for API and jupyter tutorial index

### PR DESCRIPTION
**Pull request by 317brian**
* Adds an index page that links to the notebook. Index page includes prereqs for the notebooks
* Updates gitignore so that checkpoint files are not added to GitHub
* The notebook is stored in the `assets` folder. This shouldn't be an issue since it's not much bigger than a markdown file of equivalent length.

After this is reviewed/approved, I will make a commit that locks the markdown cells to be uneditable by default. (This is more a guardrail for accidents. Someone still could if they change the metadata).

### CODE REVIEW

File: docs/tutorials/jupyter-index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+- Jupyter lab (recommended) or Jupyter notebook running on a non-default port. By default, Druid and Jupyter both try to use port `8888,` so start Jupyter on a different port. For example, use the following command to start Jupyter lab on port `3001`:
+   
+   ```bash
+   jupyter lab --port 3001
+   ```
+
+- An available Druid instance. You can use the `micro-quickstart` configuration described in [Quickstart (local)](./index.md). The tutorials assume that you are using the quickstart, so no authentication or authorization is expected unless explicitly mentioned.
+
+- The `requests` package for Python. For example, you can install it with the following command:
```
Comment (techdocsmith) [inline docs/tutorials/jupyter-index.md:39]: Python 3?

File: docs/tutorials/jupyter-index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+There are  Jupyter notebook-based tutorials that are available for you to try as you learn more about Druid. These tutorials provide snippets of Python code that you can use to run calls against the Druid API.
```
Comment (techdocsmith) [inline docs/tutorials/jupyter-index.md:25]:
```suggestion
There are Jupyter notebook-based tutorials that are available for you to try as you learn more about Druid. These tutorials provide snippets of Python code that you can use to run calls against the Druid API.
```

Comment (techdocsmith) [reply docs/tutorials/jupyter-index.md:25]: Something I don't like about "There are". What about: You can try out the APIs using the notebooks... or something.

**Review by techdocsmith — COMMENTED** [review]
Couple comments. But otherwise this is what I was thinking.

File: docs/tutorials/jupyter-index.md (hunk 2 above)
Comment (317brian) [reply docs/tutorials/jupyter-index.md:25]:
```suggestion
You can try out the Druid APIs using the Jupyter notebook-based tutorials that are available. These tutorials provide snippets of Python code that you can use to run calls against the Druid API.
```

**Pull request merged**